src/main/java/com/amihaiemil/camel/ReadPointedScalar.java

```diff
@@ -0,0 +1,149 @@
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+/**
+ * Read Yaml Scalar when previous yaml line ends with '>' character.
+ * @author Sherif Waly (sherifwaly95@gmail.com)
+ * @version $Id$
+ * @since 1.0.2
+ *
+ */
+final class ReadPointedScalar implements YamlNode {
+
+    /**
+     * Lines to be represented as a wrapped scalar.
+     */
+    private AbstractYamlLines lines;
+    
+    /**
+     * Ctor.
+     * @param lines Given lines to represent.
+     */
+    ReadPointedScalar(final AbstractYamlLines lines) {
+        this.lines = lines;
+    }
+   
+    @Override
+    public int compareTo(final YamlNode other) {
+        int result = -1;
+        if (this == other) {
+            result = 0;
+        } else if (other == null) {
+            result = 1;
+        } else if (other instanceof Scalar) {
+            result = this.value().compareTo(((Scalar) other).value());
+        } else if (other instanceof ReadPipeScalar) {
+            result = this.value().compareTo(((ReadPipeScalar) other).value());
+        } else if (other instanceof ReadPointedScalar) {
+            result =
+                this.value().compareTo(((ReadPointedScalar) other).value());
+        }
+        return result;
+    }
+
+    @Override
+    public Collection<YamlNode> children() {
+        return new LinkedList<YamlNode>();
+    }
+
+    @Override
+    public String indent(final int indentation) {
+        StringBuilder builder = new StringBuilder();
+        for(final YamlLine line: this.lines) {
+            if(line.trimmed().length() == 0 || line.indentation() > 0) {
+                if(this.doNotEndWithNewLine(builder)) {
+                    builder.append('\n');
+                }
+                int spaces = line.indentation();
+                if(spaces > 0) {
+                    for(int i = 0; i < spaces + indentation; i++) {
+                        builder.append(' ');
+                    }
+                }
+                builder.append(line.trimmed());
+                builder.append('\n');
+            } else {
+                if(this.doNotEndWithNewLine(builder)) {
+                    builder.append(' ');
+                } else {
+                    for(int i = 0; i < indentation; i++) {
+                        builder.append(' ');
+                    }
+                }
+                builder.append(line.trimmed());
+            }
+        }
+        return builder.toString();
+    }
+    
+    /**
+     * Checks whether StringBuilder do not end with newline or not.
+     * @param builder StringBuilder
+     * @return Boolean Whether builder do not end with newline char or not
+     */
+    private boolean doNotEndWithNewLine(final StringBuilder builder) {
+        return builder.length() > 0 
+                && builder.charAt(builder.length()-1) != '\n';
+    }
+    /**
+     * Value of this scalar.
+     * @return String
+     */
+    public String value() {
+        StringBuilder builder = new StringBuilder();
```

Comment:
@SherifWaly are you sure this algorithm is correct? It seems too complicated to me.
The requirement is: ``newlines are ignored (turned into spaces), unless it is at the end of an empty line, or a more intended line``

All that needs to be done is: iterate over the lines and concatenate the ones that are at the same indentation, **but** with the condition that their numbers are consecutive. So basically, this line number constraint is the only difference from the other wrapped scalar (piped). 

With the other lines, which are empty or more indented, you don't have to do anything, their line endings stay the same. And the same indentation lines' numbers have to be consecutive, because you don't want to concatenate lines which have nested nodes/empty lines between them:

```yaml
>
  Sammy Sosa completed another (0)
  fine season with great stats. (1)
(2)
    63 Home Runs (3)
    0.288 Batting Average (4)
(5)
  What a year! (6)
```
turns into 
```yaml
>
  Sammy Sosa completed another (0) fine season with great stats. (1)
(2)
    63 Home Runs (3)
    0.288 Batting Average (4)
(5)
  What a year! (6)
```
You don't concatenate lines (2) and (6) :D 

Is this what your algorithm does, or am I missing something?